lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -111,6 +116,42 @@ type AddInvoiceData struct {
 // paymentHashAndPreimage returns the payment hash and preimage for this invoice
 // depending on the configuration.
 func (d *AddInvoiceData) paymentHashAndPreimage() (*lntypes.Preimage, lntypes.Hash, error) {
+	if d.Amp {
+		return d.ampPaymentHashAndPreimage()
+	}
+
+	return d.mppPaymentHashAndPreimage()
+}
+
+// ampPaymentHashAndPreimage returns the payment hash to use for an AMP invoice.
+// The preimage will always be nil.
+func (d *AddInvoiceData) ampPaymentHashAndPreimage() (*lntypes.Preimage, lntypes.Hash, error) {
+	switch {
+
+	// Preimages cannot be set on AMP invoice.
+	case d.Preimage != nil:
+		return nil, lntypes.Hash{},
+			errors.New("preimage set on AMP invoice")
+
+	// If a specific hash was requested, use that.
+	case d.Hash != nil:
+		return nil, *d.Hash, nil
```

Comment:
Why would you request a specific hash on an AMP invoice?

Reply:
users could choose to use a specific identifier for the invoice, but it's not necessary. you think it is better to just to not permit this?

Reply:
If this is exposed on the API, I indeed think it is better to not permit it. To reduce the level of confusion about what AMP is and also not to get applications that use a feature that may disappear in the future.